Maps (#5414)
SkylineCommunications/dataminer-docs: dataminer/Functions/Dashboards_and_Low_Code_Apps/Visualizations/Available_visualizations/General/Maps.md

# Maps

Available from DataMiner 10.4.0 [CU13]/10.5.0 [CU1]/DataMiner 10.5.4 onwards<!-- RN 42309 -->.

The maps component lets you **visualize and explore geospatial data using markers and/or lines**. It currently only supports Google Maps ("gmaps") as the [Maps provider](xref:Configuring_the_DataMiner_Maps_host_servers).

![Maps](~/dataminer/images/Maps_Component.png)<br>*Maps component in DataMiner 10.4.4*

With this component, you can:

- Track both real-time and [historical location data](#using-the-map-for-history-tracking) to monitor movements or analyze trends.

- [Select markers or lines](#using-the-map-as-a-selector) on the map to access more details.

- See your data in a way that makes sense, thanks to a range of [customization options](#customizing-markers).

- Toggle [extra layers on demand](#displaying-extra-layers) to add context without cluttering the map.

## Prerequisites

- To display the map, the machine running the browser must have internet access.

> - To use the maps component, the host servers for DataMiner Maps have to be configured in the file `C:\Skyline DataMiner\Maps\ServerConfig.xml`. If this file does not exist, it will be created automatically when you use a maps component for the first time. To change the configuration, see [Configuring the DataMiner Maps host servers](xref:Configuring_the_DataMiner_Maps_host_servers).

Author: ```suggestion
- To use the maps component, a valid maps configuration is required. See [Configuring the DataMiner Maps host servers](xref:Configuring_the_DataMiner_Maps_host_servers).
```

## Supported data types

The maps component is used to display the results of queries in a map format. It should therefore **always be configured with [query data input](xref:Query_Data)**.

Each row in a query corresponds to a marker or line on the map.

## Interacting with the maps component

> In read mode, you can interact with the maps component in several ways to explore and customize the map view.

Reviewer: What is the "Read mode" exactly? Is that when not editing the dashboard/app? If so, I wouldn't even mention it.

- Hover over the specific area you want to zoom in or out from and scroll up or down. Depending on your DataMiner version and the [*Hold Ctrl to zoom* option](#maps-settings), you may need to hold the Ctrl key while scrolling.

### Panning

Panning allows you to **move the visible area of the map**. You can pan in the following ways:

- Use one of the following keyboard shortcuts:

  | Keyboard shortcut | Action |
  |--|--|
  | ← | Move left |
  | → | Move right |
  | ↑ | Move up |
  | ↓ | Move down |
  | Home | Jump left by 75% |
  | End | Jump right by 75% |
  | Page Up | Jump up by 75% |
  | Page Down | Jump down by 75% |

- Click and hold the scroll wheel anywhere on the map and drag the mouse pointer in the desired direction.

  ![Panning](~/dataminer/images/Maps_Panning.gif)<br>*Maps component in DataMiner 10.5.11*

### Selecting which layers are shown

In the top-left corner of the component, you can choose which layers are displayed on the map by selecting them from the dropdown menu.

![Displaying different layers](~/dataminer/images/Maps_Layers.gif)<br>*Maps component in DataMiner 10.5.11*

### Changing the map type

> When the [*Show map type control* setting](#map-settings) is enabled, you can change the type of map directly in read mode.

Reviewer: Wouldn't mention "read mode".

In the lower-left corner of the map, you can **switch between the supported map types** using the dropdown menu. The available map types depend on the map provider. At present, only Google Maps is supported, with the following types:

- *roadmap*

- *satellite*

- *hybrid*

- *terrain*

![Map types](~/dataminer/images/Map_Types.gif)<br>*Maps component in DataMiner 10.5.11*

The [*Default map type* setting](#map-settings) determines which type of map is displayed initially when the dashboard or app is opened.

## Displaying extra layers

To provide additional context or details on a map without cluttering the base view, you can **configure a button to display or hide an extra layer** on demand:

1. Add a button component to the dashboard or page.

1. Configure an on-click action:

   1. In the *Settings* pane, click the *Configure actions* button next to *On click*.

   1. From the dropdown list, select *Execute component action*.

   1. Specify the following details:

      - *I want to*: `Maps, Toggle overlay`

>       - *Which form?*: Select the desired maps component.

Author: ```suggestion
      - *Which map?*: Select the desired maps component.
```
